bigtable/google/cloud/bigtable/instance.py

```diff
@@ -282,3 +286,189 @@ def list_tables(self):
+                           description='', ignore_warnings=None,
+                           cluster_id=None, allow_transactional_writes=False):
+        """Creates an app profile within an instance.
+
+        :type: app_profile_id: str
+        :param app_profile_id: The unique name for the new app profile.
+
+        :type: routing_policy_type: int
+        :param: routing_policy_type: There are two routing policies
+                                    ROUTING_POLICY_TYPE_ANY = 1 and
+                                    ROUTING_POLICY_TYPE_SINGLE = 2.
+                                    By default it is ROUTING_POLICY_TYPE_ANY
+                                    which will create a
+                                    MultiClusterRoutingUseAny policy.
+                                    If ROUTING_POLICY_TYPE_ANY is specified, a
+                                    SingleClusterRouting policy will be created
+                                    using the cluster_id and
+                                    allow_transactional_writes parameters.
+
+        :type: description: str
+        :param: description: (Optional) Long form description of the use
+                                case for this AppProfile.
+
+        :type: ignore_warnings: bool
+        :param: ignore_warnings: (Optional) If true, ignore safety checks when
+                                    creating the app profile.
+
+        :type: cluster_id: str
+        :param: cluster_id: (Optional) Unique cluster_id which is only required
+                            when routing_policy_type is
+                            ROUTING_POLICY_TYPE_SINGLE.
+
+        :type: allow_transactional_writes: bool
+        :param: allow_transactional_writes: (Optional) If true, allow
+                                            transactional writes for
+                                            ROUTING_POLICY_TYPE_SINGLE.
+
+        :rtype: :class:`~google.cloud.bigtable_admin_v2.types.AppProfile`
+        :return: The AppProfile instance.
+        """
+        single_cluster_routing = None
+        multi_cluster_routing_use_any = None
+        instance_admin_client = self._client._instance_admin_client
+        name = instance_admin_client.app_profile_path(
+            self._client.project, self.name, app_profile_id)
+
+        if routing_policy_type == ROUTING_POLICY_TYPE_ANY:
+            multi_cluster_routing_use_any = (
+                instance_pb2.AppProfile.MultiClusterRoutingUseAny())
+
+        if routing_policy_type == ROUTING_POLICY_TYPE_SINGLE:
+            single_cluster_routing = (
+                instance_pb2.AppProfile.SingleClusterRouting(
+                    cluster_id=cluster_id,
+                    allow_transactional_writes=allow_transactional_writes
+                ))
+
+        app_profile = instance_pb2.AppProfile(
+            name=name, description=description,
+            multi_cluster_routing_use_any=multi_cluster_routing_use_any,
```

Comment:
In Java protobufs, one cannot set both multi and single cluster routing policies at the same time.  Please add a system test for both cases (single and multi cluster).